libnavui-dropin/src/main/java/com/mapbox/navigation/dropin/DropInNavigationViewContext.kt

```diff
@@ -25,4 +42,60 @@ internal class DropInNavigationViewContext(
     var routeArrowOptions: RouteArrowOptions = RouteArrowOptions.Builder(context)
         .withAboveLayerId(RouteLayerConstants.TOP_LEVEL_ROUTE_LINE_LAYER_ID)
         .build()
+
+    val onBackPressedCallback = object : OnBackPressedCallback(true) {
+        override fun handleOnBackPressed() = viewModel.onBackPressed()
+    }
+
+    val navigation: MapboxNavigation get() = checkNotNull(MapboxNavigationApp.current()) {
+        "MapboxNavigationApp not initialized. You must call MapboxNavigationApp.setup() first."
+    }
+
+    //region Builders & Factories
+
+    fun routeOptionsBuilder() = RouteOptions.builder()
+        .applyDefaultNavigationOptions()
+        .applyLanguageAndVoiceUnitOptions(context)
+
+    fun mapAnnotationFactory() = MapMarkerFactory(
+        context,
+        BitmapMemoryCache(4 * MB_IN_BYTES)
+    )
+
+    //endregion
+
+    //region Use Cases
+
+    fun fetchRoutesUseCase() = FetchRouteUseCase(
+        navigation,
+        this::routeOptionsBuilder,
+        getCurrentLocationUseCase(),
+        Dispatchers.IO
+    )
+
+    fun fetchAndSetRouteUseCase() = FetchAndSetRouteUseCase(
+        navigation,
+        fetchRoutesUseCase(),
+        Dispatchers.Main
+    )
+
+    fun getCurrentLocationUseCase() = GetCurrentLocationUseCase(
+        navigation,
+        viewModel.locationBehavior,
+        Dispatchers.Default
+    )
+
+    fun startActiveGuidanceUseCase() = StartActiveGuidanceUseCase(
+        navigation,
+        viewModel._activeNavigationStarted,
+        Dispatchers.Main
+    )
+
+    fun stopActiveGuidanceUseCase() = StopActiveGuidanceUseCase(
+        navigation,
+        viewModel._activeNavigationStarted,
+        Dispatchers.Main
+    )
```

Comment:
This is a ton of boilerplate code that can be simplified with Dagger 2. What do you guys think?

Reply:
See https://github.com/mapbox/mapbox-navigation-android/pull/5517